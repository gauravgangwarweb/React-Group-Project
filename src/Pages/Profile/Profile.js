import React from 'react';
import { useSelector } from 'react-redux';

const MyProfile = () => {
  const missionsState = useSelector((state) => state.missionsReducer.missions);
  const joinedMissions = missionsState.filter(
    (missions) => missions.reserved === true,
  );

  return (
    <div className="missions-container">
      <h2 className="missions-header">My Missions</h2>
      <ul className="missions-list">
        {joinedMissions.map((mission) => (
          <li key={mission.mission_id}>{mission.mission_name}</li>
        ))}
      </ul>
    </div>
  );
};

export default MyProfile;
