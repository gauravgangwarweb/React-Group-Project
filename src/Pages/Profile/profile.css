.profile-container {
  width: 95%;
  box-shadow: rgba(0, 0, 0, 0.1) 0 4px 12px;
  height: 660px;
}

.profile-heading {
  text-align: center;
  font-family: 'Aboreto', cursive;
  font-weight: 800;
  font-size: 30px;
  color: #f68d36;
}

.main-container {
  display: flex;
  justify-content: center;
  width: 100%;
  gap: 25px;
}

.rockets-container {
  width: 40%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  box-shadow: rgba(60, 64, 67, 0.3) 0 1px 2px 0, rgba(60, 64, 67, 0.15) 0 1px 3px 1px;
  padding: 0 10px 0 10px;
}

.missions-container {
  width: 40%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  box-shadow: rgba(60, 64, 67, 0.3) 0 1px 2px 0, rgba(60, 64, 67, 0.15) 0 1px 3px 1px;
  padding: 0 10px 0 10px;
}

h3 {
  color: #c676f6;
  font-size: 22px;
  width: 100%;
  text-align: center;
  border-bottom: 2px solid #d70e5b;
  padding-bottom: 15px;
}

ul {
  display: flex;
  gap: 15px;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 450px;
}

.container-span {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 70%;
  height: 50px;
  font-size: 20px;
  font-weight: 300;
  text-align: center;
  box-shadow: rgba(9, 30, 66, 0.25) 0 4px 8px -2px, rgba(9, 30, 66, 0.08) 0 0 0 1px;
  list-style: none;
}
