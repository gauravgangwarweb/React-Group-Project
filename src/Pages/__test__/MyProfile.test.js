import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../Redux/consfigureStore';
import Profile from '../Profile/Profile';

const ProfileProvider = () => (
  <Provider store={store}>
    <Profile />
  </Provider>
);

describe('test profile component', () => {
  it('render profile correctly', () => {
    const Profile = renderer.create(<ProfileProvider />);
    expect(Profile).toMatchSnapshot();
  });
});
