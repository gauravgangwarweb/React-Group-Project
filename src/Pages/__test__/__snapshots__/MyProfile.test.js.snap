// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test profile component render profile correctly 1`] = `
<div
  className="profile-container"
>
  <h2
    className="profile-heading"
  >
    My Profile
  </h2>
  <div
    className="main-container"
  >
    <div
      className="rockets-container"
    >
      <h3>
        My Rockets
      </h3>
      <ul />
    </div>
    <div
      className="missions-container"
    >
      <h3>
        My Missions
      </h3>
      <ul />
    </div>
  </div>
</div>
`;
